### PR TITLE
Feature: Node: setMaintanceMode

### PR DESCRIPTION
**Pull request by DerMistkaefer**
This PR introduces the functionality to set a Node into Maintenance Mode.

API Doc:
https://dashflo.net/docs/api/pterodactyl/v1/
image

### CODE REVIEW

**Comment by lokerhp** [comment]
Personally I would change hasMaintanceMode into isInMaintenanceMode or inMaintenanceMode or something like that

**Comment by DerMistkaefer** [comment]
Thank you for your suggestion regarding the naming convention.
Based on your feedback, I have updated the property to a more descriptive name, `isInMaintenanceMode`.
However, I have decided to keep the function name as `setMaintanceMode` to maintain consistency with the existing getter `hasMaintanceMode`, as referenced in [Entity Node L45](https://github.com/mattmalec/Pterodactyl4J/blob/develop/src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java#L45). 
This balance allows us to improve clarity while preserving coherence in our codebase.

File: src/main/java/com/mattmalec/pterodactyl4j/requests/action/AbstractNodeAction.java [hunk 1]
```diff
@@ -128,4 +129,10 @@ public NodeAction setScheme(boolean secure) {
 		this.secure = secure;
 		return this;
 	}
+
+	@Override
+	public NodeAction setMaintanceMode(boolean isInMaintenanceMode) {
```
Comment (mattmalec) [inline src/main/java/com/mattmalec/pterodactyl4j/requests/action/AbstractNodeAction.java]: we should deprecate this method and make a new method that spells maintenance correctly. the old method can just call the new one

File: src/main/java/com/mattmalec/pterodactyl4j/application/managers/NodeAction.java [hunk 2]
```diff
@@ -65,4 +65,6 @@ default NodeAction setDiskSpaceOverallocate(long diskSpaceOverallocate) {
 	NodeAction setThrottle(boolean throttle);
 
 	NodeAction setScheme(boolean secure);
+
+	NodeAction setMaintanceMode(boolean isInMaintenanceMode);
```
Comment (mattmalec) [inline src/main/java/com/mattmalec/pterodactyl4j/application/managers/NodeAction.java]: think of writing getters as if you're asking a question. in our case, asking if the node is in maintenance mode is correctly defined as `isInMaintenanceMode`. however, for setters, this logic generally doesn't apply - the variable name should be prescriptive instead.

i think naming this `maintenanceMode` is more than okay. we should only use `isInMaintenanceMode` for the getter, but internally pass `maintenanceMode` around

File: src/main/java/com/mattmalec/pterodactyl4j/requests/action/AbstractNodeAction.java (hunk 1 above)
Comment (mattmalec) [reply src/main/java/com/mattmalec/pterodactyl4j/requests/action/AbstractNodeAction.java]: also see above comments on variable naming

**Review by mattmalec — CHANGES_REQUESTED** [review]
hey there! thanks so much for your contribution to p4j! just some small nitty changes, and then it should be good to go :)

**Comment by DerMistkaefer** [comment]
Thank you for your feedback. I don't want to change the old function name but when you say it is okay then let's go.
Now we use correct naming.

File: src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java [hunk 3]
```diff
@@ -42,7 +42,16 @@ public interface Node extends ISnowflake {
 
 	boolean isBehindProxy();
 
-	boolean hasMaintanceMode();
+	boolean isInMaintenanceMode();
+
+	/**
+	 * @deprecated Use {@link #isInMaintenanceMode()} instead
```
Comment (mattmalec) [inline src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java]: docstrings should be formatted [like this](https://github.com/mattmalec/Pterodactyl4J/blob/develop/src/main/java/com/mattmalec/pterodactyl4j/client/entities/PteroClient.java#L38-L49)

Comment (mattmalec) [reply src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java]: also make sure to clarify that it's only deprecated to fix the spelling error

**Comment by DerMistkaefer** [comment]
@mattmalec can we please merge this?

**Review by mattmalec — CHANGES_REQUESTED** [review]
super quick changes and i will approve + merge today

File: src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java (hunk 3 above)
Comment (DerMistkaefer) [reply src/main/java/com/mattmalec/pterodactyl4j/application/entities/Node.java]: Okay I fixed the ranking in den docstring and add a notice about the reason of the deprecation.

**Review by mattmalec — APPROVED** [review]
looks good!

**Pull request merged**